tfhe/src/integer/server_key/radix_parallel/add.rs

```diff
@@ -282,8 +282,12 @@ impl ServerKey {
         lhs: &mut RadixCiphertext,
         rhs: &RadixCiphertext,
     ) {
-        debug_assert!(lhs.block_carries_are_empty());
-        debug_assert!(rhs.block_carries_are_empty());
+        let degree_after_add_does_not_go_beyond_first_carry =
+            lhs.blocks.iter().zip(rhs.blocks.iter()).all(|(bl, br)| {
+                let degree_after_add = bl.degree.0 + br.degree.0;
+                degree_after_add < (self.key.message_modulus.0 * 2)
```

Comment:
should be message_modulus * carry_modulus ?

Reply:
No, it would not be the same `message_modulus * carry_modulus` is the whole plaintext space, here we want to not go beyond first carry

Reply:
ah yes indeed ! thanks